packages/javascript/bh-shared-ui/src/utils/array.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright 2025 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * Returns true if array `a` and `b` contain the same elements in any order
+ *
+ * Uses referential comparison by default.
+ * For non-primitive arrays, provide a consistent, collision-free hash function.
+ */
+export const areArraysSimilar = <T>(a: T[], b: T[], hashFn?: (i: T) => string | number) => {
```

Comment:
If you saw the previous version, this has been updated. It no longer requires a sort method. Not only does this make the API simpler, it makes the method run in `O(n)` which is much better than the previous `O(n log n)`